 import java.util.HashMap;
class Solution {
    public int majorityElement(int[] nums) {
        // import java.util.HashMap;
        // import java.util.HashMap
            HashMap<Integer,Integer>map=new HashMap<>();
            for(int i=0;i<nums.length;i++)
            {
                if(map.containsKey(nums[i]))
                {
                    map.put(nums[i],map.get(nums[i])+1);
                }
                else{
                    map.put(nums[i],1);
                }
            }
            int n=nums.length/2;
            for(int i=0;i<nums.length;i++)
            {
                if(map.get(nums[i])>n)
                {
                return nums[i];
                }
            
            }
            return-1;
    }
}
 